mitmproxy uninstallable in NixOS 20.09

**Describe the bug**
Since upgrading to NixOS 20.09, I cannot install mitmproxy.

**To Reproduce**
With the nixos-20.09 channel
```
$ nix-channel --list
nixos-20.09 https://nixos.org/channels/nixos-20.09

$ nix-shell -p mitmproxy
```

**Expected behavior**
Package is installed.

**Screenshots**
Package fails to install.  Tail of output is as follows:
```
=========================== short test summary info ============================
FAILED test/mitmproxy/net/test_tcp.py::TestSSLUpstreamCertVerificationWBadHostname::test_mode_none_should_pass_without_sni
FAILED test/mitmproxy/net/test_tcp.py::TestSSLUpstreamCertVerificationWValidCertChain::test_mode_strict_w_pemfile_should_pass
FAILED test/mitmproxy/net/test_tcp.py::TestSSLUpstreamCertVerificationWValidCertChain::test_mode_strict_w_confdir_should_pass
FAILED test/mitmproxy/proxy/test_server.py::TestHTTPSUpstreamServerVerificationWTrustedCert::test_verification_w_confdir
FAILED test/mitmproxy/proxy/test_server.py::TestHTTPSUpstreamServerVerificationWTrustedCert::test_verification_w_pemfile
= 5 failed, 1484 passed, 1 deselected, 2 xfailed, 103 warnings in 93.92s (0:01:33) =
builder for '/nix/store/p2pw9y8jsdyll8c1rvsm2wq0yfdk08gn-python3.8-mitmproxy-5.2.drv' failed with exit code 1
error: build of '/nix/store/p2pw9y8jsdyll8c1rvsm2wq0yfdk08gn-python3.8-mitmproxy-5.2.drv' failed
```

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
@fpletz @kamilchm

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
aji$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/wpfm6lk7dyiafvwa6n3s9dk62yv9h5sk-nix-info
copying path '/nix/store/wpfm6lk7dyiafvwa6n3s9dk62yv9h5sk-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.72, NixOS, 20.09beta1346.05334ad7852 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(sjg): `"nixos-20.09-20.09.1819.f41c0fc7d8e, nixpkgs-21.03pre251181.dd1b7e377f6"`
 - channels(root): `"nixos-20.09beta1346.05334ad7852"`
 - nixpkgs: `/home/sjg/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

Have same failures in unstable channel
```
E               mitmproxy.exceptions.InvalidCertificateException: Certificate verification error for example.mitmproxy.org: self signed certificate (errno: 18, depth: 0)

mitmproxy/net/tcp.py:399: InvalidCertificateException
_ TestHTTPSUpstreamServerVerificationWTrustedCert.test_verification_w_confdir __

...
>           assert "Cannot validate hostname, SNI missing." in str(c.ssl_verification_error)
E           AssertionError: assert 'Cannot validate hostname, SNI missing.' in 'Certificate verification error for None: self signed certificate (errno: 18, depth: 0)'
E            +  where 'Certificate verification error for None: self signed certificate (errno: 18, depth: 0)' = str(InvalidCertificateException('Certificate verification error for None: self signed certificate (errno: 18, depth: 0)'))
E            +    where InvalidCertificateException('Certificate verification error for None: self signed certificate (errno: 18, depth: 0)') = <mitmproxy.net.tcp.TCPClient object at 0x7fffe4149d00>.ssl_verification_error

test/mitmproxy/net/test_tcp.py:303: AssertionError
_ TestSSLUpstreamCertVerificationWValidCertChain.test_mode_strict_w_pemfile_should_pass _
...
>       raise exception_type(errors)
E       OpenSSL.SSL.Error: [('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')]

/nix/store/6nc0npzz9sxqczgxx6vy87x9wnm4gfxl-python3.8-pyOpenSSL-19.1.0/lib/python3.8/site-packages/OpenSSL/_util.py:54: Error

During handling of the above exception, another exception occurred:

self = <test.mitmproxy.net.test_tcp.TestSSLUpstreamCertVerificationWValidCertChain object at 0x7fffe435ad00>
tdata = <mitmproxy.utils.data.Data object at 0x7fffe435ad90>

    def test_mode_strict_w_confdir_should_pass(self, tdata):
        c = tcp.TCPClient(("127.0.0.1", self.port))
        with c.connect():
>           c.convert_to_tls(
                sni="example.mitmproxy.org",
                verify=SSL.VERIFY_PEER,
                ca_path=cdata.path("data/verificationcerts/")
            )

test/mitmproxy/net/test_tcp.py:344:
```

--- a/pkgs/development/python-modules/mitmproxy/default.nix
+++ b/pkgs/development/python-modules/mitmproxy/default.nix
@@ -37,32 +37,25 @@
 , parver
 , pytest-asyncio
 , hypothesis
+, asgiref
+, msgpack
 }:
 
 buildPythonPackage rec {
   pname = "mitmproxy";
-  version = "5.2";
+  version = "5.3.0";
   disabled = isPy27;
 
   src = fetchFromGitHub {
     owner  = pname;
     repo   = pname;
     rev    = "v${version}";
-    sha256 = "0ja0aqnfmkvns5qmd51hmrvbw8dnccaks30gxgzgcjgy30rj4brq";
+    sha256 = "04y7fxxssrs14i7zl7fwlwrpnms39i7a6m18481sg8vlrkbagxjr";
   };
-
-  patches = [
-    # Apply patch from upstream to make mitmproxy v5.2 compatible with urwid >v2.1.0
-    (fetchpatch {
-      name = "urwid-lt-2.1.0.patch";
-      url = "https://github.com/mitmproxy/mitmproxy/commit/ea9177217208fdf642ffc54f6b1f6507a199350c.patch";
-      sha256 = "1z5r8izg5nvay01ywl3xc6in1vjfi9f144j057p3k5rzfliv49gg";
-    })
-  ];
 
   postPatch = ''
     # remove dependency constraints
-    sed 's/>=\([0-9]\.\?\)\+\( \?, \?<\([0-9]\.\?\)\+\)\?//' -i setup.py
+    sed 's/>=\([0-9]\.\?\)\+\( \?, \?<\([0-9]\.\?\)\+\)\?\( \?, \?!=\([0-9]\.\?\)\+\)\?//' -i setup.py
   '';
 
   doCheck = (!stdenv.isDarwin);
@@ -75,6 +68,7 @@ buildPythonPackage rec {
   propagatedBuildInputs = [
     setuptools
     # setup.py
+    asgiref
     blinker
     brotli
     certifi
@@ -85,6 +79,7 @@ buildPythonPackage rec {
     hyperframe
     kaitaistruct
     ldap3
+    msgpack
     passlib
     protobuf
     publicsuffix2

